src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -640,7 +645,9 @@ private void emitStudyChanged(String studyName, String updateType) {
     }
 
     private Mono<Void> assertSecurityAnalysisNotRunning(String studyName, String userId) {
-        return Mono.empty(); // FIXME the security analysis status is not yet implemented
+        Mono<String> statusMono = getSecurityAnalysisStatus(studyName, userId);
+        return statusMono.switchIfEmpty(Mono.empty())
+                .flatMap(s -> s.equals(SecurityAnalysisStatus.RUNNING.name()) ? Mono.error(new StudyException(SECURITY_ANALYSIS_RUNNING)) : Mono.empty());
```

Comment:
When can statusMono be empty ? (warning: flatmap doesn't execute on empty monos)

Reply:
Corrected